x-pack/plugins/index_management/public/application/sections/home/index_list/details_page/trained_models_deployment_modal.tsx

```diff
@@ -0,0 +1,180 @@
+import type { SharePluginStart } from '@kbn/share-plugin/public';
+import { i18n } from '@kbn/i18n';
+
+interface SemanticTextProps {
+  isSemanticTextEnabled: boolean;
+  setIsModalVisible: (isVisible: boolean) => void;
+  refreshModal: () => void;
+  pendingDeployments: Array<string | undefined>;
+  errorsInTrainedModelDeployment: string[];
+  url?: SharePluginStart['url'];
+}
+
+const ML_APP_LOCATOR = 'ML_APP_LOCATOR';
+const TRAINED_MODELS_MANAGE = 'trained_models';
+
+export function TrainedModelsDeploymentModal({
+  isSemanticTextEnabled,
+  setIsModalVisible,
+  refreshModal,
+  pendingDeployments = [],
+  errorsInTrainedModelDeployment = [],
+  url,
+}: SemanticTextProps) {
+  const modalTitleId = useGeneratedHtmlId();
+  const closeModal = () => setIsModalVisible(false);
+  const [mlManagementPageUrl, setMlManagementPageUrl] = useState<string>('');
+
+  useEffect(() => {
+    const mlLocator = url?.locators.get(ML_APP_LOCATOR);
+    const generateUrl = async () => {
+      if (mlLocator) {
+        const mlURL = await mlLocator.getUrl({
+          page: TRAINED_MODELS_MANAGE,
+        });
+        setMlManagementPageUrl(mlURL);
+      }
+    };
+    generateUrl();
+  }, [url]);
+
+  const ErroredDeployments = pendingDeployments.filter(
+    (deployment) => deployment !== undefined && errorsInTrainedModelDeployment.includes(deployment)
+  );
+
+  const PendingModelsDeploymentModal = () => {
+    const pendingDeploymentsList = pendingDeployments.map((deployment, index) => (
+      <li key={index}>
+        <EuiHealth textSize="xs" color="warning">
+          {deployment}
+        </EuiHealth>
+      </li>
+    ));
+
+    return (
+      <EuiConfirmModal
+        aria-labelledby={modalTitleId}
+        style={{ width: 600 }}
+        title={i18n.translate(
+          'xpack.idxMgmt.indexDetails.trainedModelsDeploymentModal.titleLabel',
+          {
+            defaultMessage: 'Models still deploying',
+          }
+        )}
+        titleProps={{ id: modalTitleId }}
+        onCancel={closeModal}
+        onConfirm={refreshModal}
+        cancelButtonText={i18n.translate(
+          'xpack.idxMgmt.indexDetails.trainedModelsDeploymentModal.cancelButtonLabel',
+          {
+            defaultMessage: 'Cancel',
+          }
+        )}
+        confirmButtonText={i18n.translate(
+          'xpack.idxMgmt.indexDetails.trainedModelsDeploymentModal.refreshButtonLabel',
+          {
+            defaultMessage: 'Refresh',
+          }
+        )}
+        defaultFocusedButton="confirm"
+        data-test-subj="trainedModelsDeploymentModal"
+      >
+        <p data-test-subj="trainedModelsDeploymentModalText">
+          {i18n.translate(
+            'xpack.idxMgmt.indexDetails.trainedModelsDeploymentModal.textAboutDeploymentsNotCompleted',
+            {
+              defaultMessage:
+                'Some fields are referencing models that have not yet completed deployment. Deployment may take a few minutes to complete.',
```

Comment:
Just my 2c about the user experience: I thought this modal was a bit confusing. If I understand correctly, the operation 'Save mappings' cannot be done because deployment is not completed yet. This was not clear at first because once I read the description, I thought 'Cancel' would cancel the deployment process, not the save mappings operations (maybe that's just me). 

Then once you hit Refresh and deployment is deployed, it automatically closes the modal without any indication that the mappings are saved successfully. Small suggestion to add some sort of visual indication.

Reply:
Thank you for pointing this out. I will discuss with Michael to understand whether we can have a better visual indication here. I will create an issue for this.